app/src/main/java/ch/epfl/sdp/game/NFCTagHelper.kt

```diff
@@ -0,0 +1,13 @@
+package ch.epfl.sdp.game
+
+import android.content.Intent
+import android.nfc.NfcAdapter
+import android.nfc.Tag
+
+object NFCTagHelper {
+    fun intentToNFCTag(intent: Intent) = intent.getParcelableExtra<Tag>(NfcAdapter.EXTRA_TAG)?.id?.toHexString()
+
+    fun ByteArray.toHexString() = this.joinToString(""){ String.format("%02X",(it.toInt() and 0xFF)) }
```

Comment:
I'm not sure what type is it in that case but if it's a byte consider this:
```suggestion
    fun ByteArray.toHexString() = this.joinToString(""){ String.format("%02X",(it.toUByte())) }
```

Reply:
toUByte() is an experimental method. I think we should avoid it unless it gives us a fondamental advantage which, in my opinion, is not the case, here.